compiler/circle-mpqsolver/src/core/DataProvider.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "DataProvider.h"
+
+using namespace mpqsolver::core;
+
+using Shape = std::vector<loco::Dimension>;
+
+H5FileDataProvider::H5FileDataProvider(const std::string &h5file) : _importer(h5file)
+{
+  _importer.importGroup("value");
+  _is_raw_data = _importer.isRawData();
+}
+
+size_t H5FileDataProvider::numSamples() const { return _importer.numData(); }
+
+uint32_t H5FileDataProvider::numInputs(uint32_t sample) const
+{
+  return static_cast<uint32_t>(_importer.numInputs(sample));
+}
+
+void H5FileDataProvider::getSampleInput(uint32_t sample, uint32_t input, InputData &data) const
+{
+  if (_is_raw_data)
+  {
+    _importer.readTensor(sample, input, data._data.data(), data._data.size());
+  }
+  else
+  {
+    loco::DataType dtype;
+    Shape shape;
```

Comment:
These are not used. If you're going to define these variable, please use them to check dtype and shape of model input.

Reply:
> These are not used. If you're going to define these variable, please use them to check dtype and shape of model input.

ok. I need to update the draft at first.

Reply:
done